libs/wingsdk/src/shared-aws/domain.ts

```diff
@@ -0,0 +1,89 @@
+import { Construct } from "constructs";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { getPlatformSpecificValues } from "../util/platform-specific";
+
+/**
+ * AWS implementation of `cloud.Domain`.
+ */
+export class Domain extends cloud.Domain {
+  /** @internal */
+  protected _hostedZoneId?: string;
+  /** @internal */
+  protected _iamCertificate?: string;
+  /** @internal */
+  protected _acmCertificateArn?: string;
+
+  constructor(scope: Construct, id: string, props: cloud.DomainProps) {
+    super(scope, id, props);
+
+    const values = getPlatformSpecificValues(
+      this.node.path,
+      "iamCertificate",
+      "acmCertificateArn",
+      "hostedZoneId"
+    );
+
+    let errors: Array<String> = new Array<String>();
```

Comment:
Can we move this error rendering code to the common utility function? Seems like it's going to be needed anywhere we have platform values. Maybe `getPlatformValues()` can receives some more info about each value like defaults and description and whether it's required or not.

Reply:
I did what you requested, @eladb.

Now, the following call:

```ts
const values = getPlatformSpecificValues(
      this.node.path,
      "iamCertificate||acmCertificateArn",
      "hostedZoneId"
    );
```

I did what you requested, Elad.

Now, the following call:

ts

const values = getPlatformSpecificValues(
      this.node.path,
      "iamCertificate||acmCertificateArn",
      "hostedZoneId"
    );

checks if the value of `hostedZoneId` was provided and if one of the certificate values (`iamCertificate` or `acmCertificateArn`) was provided.

If these conditions are not valid, an exception is generated within the `getPlatformSpecificValues` function itself, allowing it to be reused by other resources in the future.